### PR TITLE
Allow dir build option

### PR DESCRIPTION
**Pull request by gmkll**
- allow configuration with dir option alternatively to out 
- example + test
- gulp end node / no stream integration

### CODE REVIEW

**Comment by jorrit** [comment]
Thanks. Is it possible to fix the issues found by DeepScan?

**Comment by gmkll** [comment]
As it is generated, I might delete the subdir complex as a whole. But why not just exclude it (in deep-scan)?The same offending code is found in expected/complex_init.js.

File: index.js [hunk 1]
```diff
@@ -49,15 +50,22 @@ module.exports = function(opts) {
...
-    output = text;
-    sourceMapOutput = sourceMap;
-  };
+  if (opts.out) {
+    opts.out = function(text, sourceMap) {
+      output = text;
+      sourceMapOutput = sourceMap;
+    };
+  }
+  /*else if (opts.dir) {
```
Comment (jorrit) [inline index.js]: Can you remove this commented section?

File: test/main.js [hunk 2]
```diff
@@ -193,15 +195,20 @@ describe('gulp-requirejs', function() {
...
+        var test = fs.readFileSync( pathname + moduleName + '.js', 'utf8');
+        test.length.should.equal(length);
+        test.should.equal(contents);
         done();
-      } catch (e) {
-        done(e);
-      }
+      }, 1);
     });
+    /*after(function() {
```
Comment (jorrit) [inline test/main.js]: And this commented piece?

**Comment by jorrit** [comment]
It has been released in v1.3.0. Thanks!

**Pull request merged**